src/api/app/views/person/notifications/_notification.xml.builder

```diff
@@ -1,10 +1,20 @@
 builder.notification(id: notification.id) do
   builder.title notification.title
-  builder.event_type notification.event_type.split('::').last.underscore
-  builder.comment_ notification.event_payload['comment'] if notification.event_payload['comment']
-  builder.description notification.event_payload['description'] if notification.event_payload['description']
-  builder.state notification.event_payload['state'] if notification.event_payload['state']
-  builder.when notification.event_payload['when'] if notification.event_payload['when']
+  builder.type notification.event_type.split('::').last.underscore
+  builder.time notification.created_at
+  if (state = if notification.notifiable_type == 'BsRequest'
+                notification.notifiable.state
+              elsif notification.notifiable_type == 'WorkflowRun'
+                notification.notifiable.status
+              end)
+    builder.state state
+  end
```

Comment:
Too much ifs and assignments for my taste :sweat_smile: Can we make this simpler to read and understand?